File: Python/2-Medium/00016._最接近的三数之和/00016._最接近的三数之和.py
def three_sum_closest(nums, target):
    """
    寻找三个数的和，使得这个和最接近给定的目标值。

    参数:
        nums (List[int]): 整数数组。
        target (int): 目标值。

    返回:
        int: 最接近目标值的三个数的和。
    
    示例:
        >>> three_sum_closest([-1, 2, 1, -4], 1)
        2
    """
    nums.sort()  # 对数组进行排序
    closest_sum = float('inf')  # 初始化一个非常大的数表示最接近的和

    for i in range(len(nums) - 2):  # 遍历到倒数第三个数即可
        left, right = i + 1, len(nums) - 1  # 双指针分别指向i之后的第一个数和数组的最后一个数
        while left < right:
            current_sum = nums[i] + nums[left] + nums[right]  # 计算当前三个数的和
            if abs(current_sum - target) < abs(closest_sum - target):
                closest_sum = current_sum  # 更新最接近的和

            if current_sum < target:  # 如果和小于目标值，则左指针右移
                left += 1
            elif current_sum > target:  # 如果和大于目标值，则右指针左移
                right -= 1
            else:
                return current_sum  # 如果等于目标值，直接返回这个和

    return closest_sum  # 返回最终计算出的最接近的和

# 运行代码
nums = [-1, 2, 1, -4]
target = 1
result = three_sum_closest(nums, target)
print(f"最接近的三数之和为: {result}")

File: C++/2-Medium/00016._最接近的三数之和/00016._最接近的三数之和.cpp
#include <iostream>
#include <vector>
#include <algorithm> // 用于std::sort
using namespace std;

// 函数：找出最接近的三数之和
int threeSumClosest(vector<int>& nums, int target) {
    sort(nums.begin(), nums.end()); // 对数组进行排序
    int closestSum = nums[0] + nums[1] + nums[2]; // 初始化最接近的和为前三个数的和
    for (int i = 0; i < nums.size() - 2; i++) { // 遍历数组
        int left = i + 1, right = nums.size() - 1; // 初始化左右指针
        while (left < right) { // 当左指针小于右指针时循环
            int currentSum = nums[i] + nums[left] + nums[right]; // 当前三数之和
            if (abs(target - currentSum) < abs(target - closestSum)) { // 如果当前和更接近target
                closestSum = currentSum; // 更新最接近的和
            }
            if (currentSum > target) { // 根据和与target的比较结果移动指针
                right--; // 如果当前和大于target，右指针左移
            } else if (currentSum < target) {
                left++; // 如果当前和小于target，左指针右移
            } else {
                return currentSum; // 如果等于target，直接返回当前和
            }
        }
    }
    return closestSum; // 返回最接近的三数之和
}

// 主函数
int main() {
    vector<int> nums = {-1, 2, 1, -4}; // 示例数组
    int target = 1; // 目标值
    int result = threeSumClosest(nums, target); // 调用函数计算结果
    cout << "最接近的三数之和为: " << result << endl; // 输出结果
    return 0;
}